### PR TITLE
DEX Part 2

### PR DESCRIPTION
**Pull request by jmenon**

### CODE REVIEW

File: src/ngraph/runtime/cpu/cpu_builder.cpp [hunk 1]
```diff
@@ -247,6 +288,151 @@ namespace ngraph
...
+                if (args.size() > 2)
+                {
+                    auto& arg2_tensor = tensor_data[args[2].get_name()];
+
+                    auto axes = mm->get_broadcast_axes();
+                    if (axes.size() == 1)
+                    {
+                        if (*(axes.begin()) == 0)
+                        {
+                            vector<float> ones_row(arg2_shape[0], 1.0f);
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_builder.cpp]: when does this get cleaned up?

File: src/ngraph/runtime/cpu/cpu_builder.cpp [hunk 2]
```diff
@@ -228,6 +231,44 @@ namespace ngraph
...
+                auto element_count = out[0].get_size();
+                auto& arg0_tensor = tensor_data[args[0].get_name()];
+                auto& out0_tensor = tensor_data[out[0].get_name()];
+
+                auto functor = [&, kernel, element_count](CPURuntimeContext* ctx) {
+                    kernel(arg0_tensor, out0_tensor, element_count);
+                };
+                functors.emplace_back(functor);
+            }
+
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_builder.cpp]: How about a MACRO here for defining these functions?

File: src/ngraph/runtime/cpu/cpu_builder.cpp (hunk 1 above)
Comment (jmenon) [reply src/ngraph/runtime/cpu/cpu_builder.cpp]: local one gets destroyed after end of scope, captured one when all call frames have gone out of scope

File: src/ngraph/runtime/cpu/cpu_builder.cpp (hunk 2 above)
Comment (jmenon) [reply src/ngraph/runtime/cpu/cpu_builder.cpp]: Sure, maybe a helper function even. The select kernel by type and rank macros are already too unwieldy

File: src/ngraph/runtime/cpu/cpu_builder.cpp (hunk 1 above)
Comment (jbobba) [reply src/ngraph/runtime/cpu/cpu_builder.cpp]: never mind, i missed the explicit capture.

**Comment by aprocter** [comment]
TESTNOW

**Pull request merged**